lib/useDataApi.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import unfetch from 'isomorphic-unfetch';
+
+const dataFetchReducer = (state, action) => {
+    switch (action.type) {
+        case 'FETCH_INIT':
+            return { ...state, isLoading: true, isError: false };
+        case 'FETCH_SUCCESS':
+            return {
+                ...state,
+                isLoading: false,
+                isError: false,
+                data: action.payload,
+            };
+        case 'FETCH_FAILURE':
+            return {
+                ...state,
+                isLoading: false,
+                isError: true,
+            };
+        default:
+            throw new Error();
+    }
+};
+
+const useDataApi = (url, initialData) => {
+    const [state, dispatch] = React.useReducer(dataFetchReducer, {
+        isLoading: false,
+        isError: false,
+        data: initialData,
+    });
+
+    React.useEffect(() => {
+        let didCancel = false;
```

Comment:
Do we really need to handle cancelling of our calls?

Reply:
Not really, I can remove

Reply:
Just asking, now that it's here it might come handy anyway